src/boot/efi/linux.c

```diff
@@ -4,89 +4,197 @@
 #include <efilib.h>
 
 #include "linux.h"
+#include "macro-fundamental.h"
+#include "missing_efi.h"
 #include "util.h"
 
-#ifdef __i386__
-#define __regparm0__ __attribute__((regparm(0)))
-#else
-#define __regparm0__
-#endif
+#define LINUX_INITRD_MEDIA_GUID \
+     {0x5568e427, 0x68fc, 0x4f3d, {0xac, 0x74, 0xca, 0x55, 0x52, 0x31, 0xcc, 0x68} }
+
+/* initrd that is served to the kernel */
+static VOID *_initrd_buffer = NULL;
+/* length of the initrd */
+static UINTN _initrd_length = 0;
+/* handle of the registerd protocol */
+static EFI_HANDLE _initrd_handle = NULL;
+
+/* work-arround for GCC */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmissing-braces"
```

Comment:
I investigated further for and apparently GNU-EFI does define DevicePath length as { UINT8, UINT8 } instead of the more sensical UINT16 (as the linux kernel headers and u-boot do), I'll redo the struct and remove this pragma